lib/grape/dsl/inside_route.rb

```diff
@@ -91,19 +91,25 @@ def error!(message, status = nil, headers = nil)
       # @param url [String] The url to be redirect.
       # @param options [Hash] The options used when redirect.
       #                       :permanent, default false.
+      #                       :body, default a short message including the URL.
       def redirect(url, options = {})
-        merged_options = options.reverse_merge(permanent: false)
-        if merged_options[:permanent]
+        permanent = options.fetch(:permanent, false)
+        body_message = options.fetch(:body, nil)
+        if permanent
           status 301
+          body_message ||= "This resource has been moved permanently to #{url}"
```

Comment:
No worries - good point.

Should we add a validator to the optional message override to ensure it adheres grammatically or just open up a separate issue for that? :smile: